src/ovp_pipeline/source_signals/base.py

```
@@ -0,0 +1,104 @@
+
+``SignalProvider`` is the contract every signal source implements.
+``Signal`` and ``AuthorityScore`` are the immutable record types passed
+between provider, orchestrator, and storage (knowledge.db /
+``60-Logs/source_authority.jsonl``).
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from types import MappingProxyType
+from typing import Any, Mapping, Protocol
```

Ruff (0.15.12)

[warning] Import from `collections.abc` instead: `Mapping`

Import from `collections.abc`

(UP035)